packages/tempus/test/DIRK/Tempus_DIRKTest.cpp

```diff
@@ -322,88 +335,90 @@ TEUCHOS_UNIT_TEST(DIRK, SinCos)
+      // Store off the final solution and step size
       StepSize.push_back(dt);
-      const double L2norm = Thyra::norm_2(*xdiff);
-      ErrorNorm.push_back(L2norm);
-    }
-
-    std::ofstream ftmp("Tempus_"+RKMethod_+"_SinCos-Error.dat");
-    double error0 = 0.8*ErrorNorm[0];
-    for (int n=0; n<(int)StepSize.size(); n++) {
-      ftmp << StepSize[n]  << "   " << ErrorNorm[n] << "   "
-           << error0*(pow(StepSize[n]/StepSize[0],order)) << std::endl;
+      auto solution = Thyra::createMember(model->get_x_space());
+      Thyra::copy(*(integrator->getX()),solution.ptr());
+      solutions.push_back(solution);
+      auto solutionDot = Thyra::createMember(model->get_x_space());
+      Thyra::copy(*(integrator->getXdot()),solutionDot.ptr());
+      solutionsDot.push_back(solutionDot);
+      if (n == nTimeStepSizes-1) {  // Add exact solution last in vector.
+        StepSize.push_back(0.0);
+        auto solution = Thyra::createMember(model->get_x_space());
+        Thyra::copy(*(model->getExactSolution(time).get_x()),solution.ptr());
+        solutions.push_back(solution);
+        auto solutionDot = Thyra::createMember(model->get_x_space());
+        Thyra::copy(*(model->getExactSolution(time).get_x_dot()),
+                    solutionDot.ptr());
+        solutionsDot.push_back(solutionDot);
+      }
+    //Teuchos::TimeMonitor::summarize();
     }
-    ftmp.close();
-
-    //if (RKMethods[m] == "SDIRK 5 Stage 4th order") {
-    //  StepSize.pop_back();  StepSize.pop_back();
-    //  ErrorNorm.pop_back(); ErrorNorm.pop_back();
-    //} else if (RKMethods[m] == "SDIRK 5 Stage 5th order") {
-    //  StepSize.pop_back();  StepSize.pop_back();  StepSize.pop_back();
-    //  ErrorNorm.pop_back(); ErrorNorm.pop_back(); ErrorNorm.pop_back();
-    //}
 
     // Check the order and intercept
-    double slope = computeLinearRegressionLogLog<double>(StepSize, ErrorNorm);
-    std::cout << "  Stepper = " << RKMethods[m] << std::endl;
-    std::cout << "  =========================" << std::endl;
-    std::cout << "  Expected order: " << order << std::endl;
-    std::cout << "  Observed order: " << slope << std::endl;
-    std::cout << "  =========================" << std::endl;
-    TEST_FLOATING_EQUALITY( slope, order, 0.03 );
-    TEST_FLOATING_EQUALITY( ErrorNorm[0], RKMethodErrors[m], 5.0e-4 );
+    double xSlope = 0.0;
+    //double xDotSlope = 0.0;
+    RCP<Tempus::Stepper<double> > stepper = integrator->getStepper();
+    double order = stepper->getOrder();
+    writeOrderError("Tempus_"+RKMethod_+"_SinCos-Error.dat",
+                    stepper, StepSize,
+                    solutions,    xErrorNorm,    xSlope); //,
+                    //solutionsDot, xDotErrorNorm, xDotSlope);
+
+    TEST_FLOATING_EQUALITY( xSlope,               order, 0.01   );
+    TEST_FLOATING_EQUALITY( xErrorNorm[0], RKMethodErrors[m], 5.0e-4 );
+    //TEST_FLOATING_EQUALITY( xDotSlope,            order, 0.01   );
```

Comment:
no `xDot` tests?

Reply:
Yeah, currently our DIRK algorithms do not calculate xDot.  I know you have means to compute these.  How hard would it be to add the xDot at the workingState in DIRK::takeStep?